packages/zevm-app-contracts/contracts/disperse/Disperse.sol

```diff
@@ -36,15 +39,20 @@ contract Disperse {
         uint256 total = 0;
         for (uint256 i = 0; i < recipients.length; i++) total += values[i];
         require(token.transferFrom(msg.sender, address(this), total));
-        for (uint256 i = 0; i < recipients.length; i++) require(token.transfer(recipients[i], values[i]));
+        for (uint256 i = 0; i < recipients.length; i++) {
+            require(token.transfer(recipients[i], values[i]));
+            emit FundsDispersed(address(token), msg.sender, recipients[i], values[i]);
+        }
     }
 
     function disperseTokenSimple(
         IERC20 token,
         address[] calldata recipients,
         uint256[] calldata values
     ) external noReentrancy {
-        for (uint256 i = 0; i < recipients.length; i++)
+        for (uint256 i = 0; i < recipients.length; i++) {
             require(token.transferFrom(msg.sender, recipients[i], values[i]));
+            emit FundsDispersed(address(token), msg.sender, recipients[i], values[i]);
+        }
```

Comment:
Our airdrop is configured to pack up to 500 users per disperse, are we ok according to block/tx gas limits given that we added the event?

Reply:
Gas limit: 30000000
Gas used without event:  779291
Gas used with event: 2144297

we are good